go/test/endtoend/cluster/cluster_process.go

```diff
@@ -929,6 +929,46 @@ func (cluster *LocalProcessCluster) StreamTabletHealth(ctx context.Context, vtta
 	return responses, nil
 }
 
+// StreamTabletHealthUntil invokes a HealthStream on a local cluster Vttablet and
+// returns the responses. It waits until a certain condition is met. The amount of time to wait is an input that it takes.
+func (cluster *LocalProcessCluster) StreamTabletHealthUntil(ctx context.Context, vttablet *Vttablet, timeout time.Duration, condition func(shr *querypb.StreamHealthResponse) bool) error {
```

Comment:
There is an inherent difference between the two. `StreamTabletHealth` gets you specified number of stream responses back, which can then be processed. On the other hand `StreamTabletHealthUntil` is meant to be used where the test doesn't care if we consume more responses and only verifies if a certain check is ever true. 
The tests currently using `StreamTabletHealth` actually can't really use `StreamTabletHealthUntil` because they want to run a test on the next packet, and not an eventual packet.